backend/src/main/java/gov/cdc/usds/simplereport/service/OrganizationInitializingService.java

```diff
@@ -66,24 +74,47 @@ public void initAll() {
 		Provider savedProvider = _providerRepo.save(_props.getProvider());
 		Map<String, DeviceType> byName = _deviceTypeRepo.findAll().stream().collect(
 				Collectors.toMap(d->d.getName(), d->d));
+        Map<String, SpecimenType> specimenTypesByName = _specimenTypeRepo.findAll().stream().collect(
+                Collectors.toMap(d -> d.getName(), d -> d));
+        for (SpecimenType s : _props.getSpecimenTypes()) {
+            SpecimenType specimenType = specimenTypesByName.get(s.getName());
+            if (null == specimenType) {
+                LOG.info("Creating device {}", s.getName());
+                specimenType = _specimenTypeRepo.save(s);
+                specimenTypesByName.put(specimenType.getName(), specimenType);
+            }
+        }
 		for (DeviceType d : _props.getDeviceTypes()) {
 			DeviceType deviceType = byName.get(d.getName());
 			if (null == deviceType) {
 				LOG.info("Creating device {}", d.getName());
 				deviceType = _deviceTypeRepo.save(d);
 				byName.put(deviceType.getName(), deviceType);
 			}
+            SpecimenType defaultTypeForDevice = _specimenTypeRepo.findByTypeCode(d.getSwabType())
+                    .orElseThrow(() -> new IllegalArgumentException("No specimen type with code " + d.getSwabType()));
+            Optional<DeviceSpecimen> deviceSpecimens = _deviceSpecimenRepo.find(d, defaultTypeForDevice);
+            if (deviceSpecimens.isEmpty()) {
+                _deviceSpecimenRepo.save(new DeviceSpecimen(deviceType, defaultTypeForDevice));
+            }
 		}
-		List<DeviceType> configured= _props.getConfiguredDeviceTypeNames().stream()
-				.map(name->byName.get(name))
+
+        List<DeviceSpecimen> configured = _props.getConfiguredDeviceTypeNames().stream()
+                .map(name -> byName.get(name).getInternalId())
+                .map(_deviceSpecimenRepo::findFirstByDeviceTypeInternalIdOrderByCreatedAt)
```

Comment:
It's the same repository call, yes—the differences are mainly that this one takes a UUID not a string and that one throws an exception rather than ignoring the problem. Honestly I'm not sure if it's just that I wrote this one first? But as I'm looking back at it, I think the better question for me to be asking is "why are we looking things up in the database when we had them in local variables in the previous block?" So I think I'll just use those instead. Thanks for pointing out the over-engineering!